roles/openshift_master_facts/tasks/main.yml

```diff
@@ -73,3 +73,9 @@
       oauth_templates: "{{ openshift_master_oauth_templates | default(None) }}"
       oauth_always_show_provider_selection: "{{ openshift_master_oauth_always_show_provider_selection | default(None) }}"
       image_policy_config: "{{ openshift_master_image_policy_config | default(None) }}"
+      builddefaults_http_proxy: "{{ openshift_builddefaults_http_proxy | default(None) }}"
+      builddefaults_https_proxy: "{{ openshift_builddefaults_https_proxy | default(None) }}"
+      builddefaults_no_proxy: "{{ openshift_builddefaults_no_proxy | default(None) }}"
+      builddefaults_git_http_proxy: "{{ openshift_builddefaults_git_http_proxy | default(None) }}"
+      builddefaults_git_https_proxy: "{{ openshift_builddefaults_git_https_proxy | default(None) }}"
+      builddefaults_config: "{{ openshift_builddefaults_json | default(builddefaults_yaml) }}"
```

Comment:
Is there a benefit to overriding the builddefaults_config here vs setting the other variables?

I suspect there might be non-proxy options that could be passed to the builddefaults_config, in which case the user provided config would be expected to override the other values. In that case, I would expect it to default to None rather than the value of builddefaults_yaml.

It also seems like there could be a case for a 'builddefaults' role that would allow for setting the individual variables. I'm wondering if it could also leverage openshift_env to make things a bit cleaner...  @abutcher thoughts?